homeassistant/components/tradfri/fan.py

```diff
@@ -147,7 +143,7 @@ async def async_turn_on(
             return
 
         if percentage is not None:
-            await self._api(self._device_control.set_mode(_from_percentage(percentage)))
+            await self._api(self._device_control.set_mode(_from_fan_speed(percentage)))
```

Comment:
We removed `set_mode` in 8.0.0, right?

Reply:
Maybe we can simply call `self.async_set_percentage` here?

Reply:
Good point, I've changed to `self.async_set_percentage`